### PR TITLE
add sample fuzzing tests

### PR DESCRIPTION
**Pull request by deads2k**
Make fuzzing tests as simple as possible from both the API installer and the scheme, so its easy to add for api groups and so that I can build a scheme and then make sure I got it right.

@kubernetes/sig-api-machinery-pr-reviews @sttts @mikedanese

### CODE REVIEW

**Comment by sttts** [comment]
lgtm

File: cmd/kubeadm/app/apis/kubeadm/serialization_test.go [hunk 1]
```diff
@@ -1,47 +0,0 @@
...
-	"k8s.io/apimachinery/pkg/runtime/serializer"
-	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/install"
-	kapitesting "k8s.io/kubernetes/pkg/api/testing"
-)
-
-const (
-	seed = 1
-)
-
-func TestRoundTripTypes(t *testing.T) {
```
Comment (deads2k) [inline cmd/kubeadm/app/apis/kubeadm/serialization_test.go]: @mikedanese this is all now a one-liner in the install package unit tests.

File: cmd/kubeadm/app/apis/kubeadm/register.go [hunk 2]
```diff
@@ -50,7 +50,3 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 	)
 	return nil
 }
-
-func (obj *MasterConfiguration) GetObjectKind() schema.ObjectKind { return &obj.TypeMeta }
```
Comment (deads2k) [inline cmd/kubeadm/app/apis/kubeadm/register.go]: These are unnecessary

File: cmd/kubeadm/app/apis/kubeadm/install/install_test.go [hunk 3]
```diff
@@ -14,16 +14,15 @@ See the License for the specific language governing permissions and
...
 import (
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/pkg/api"
+	"testing"
+
+	apitesting "k8s.io/apimachinery/pkg/api/testing"
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 )
 
-func addConversionFuncs(scheme *runtime.Scheme) error {
```
Comment (deads2k) [inline cmd/kubeadm/app/apis/kubeadm/install/install_test.go]: this is a weird diff, but metav1 now registers these automatically in your scheme when you include it.

**Comment by deads2k** [comment]
@k8s-bot kubemark e2e test this

**Pull request merged**